File: ToC_2026/8/question4/Java/2.java
import java.util.Scanner;

public class Q2_Eq {
    static boolean iszero(int m) { return m == 0; }
    static int pi(int m) { return m == 0 ? 0 : m - 1; }

    static boolean eq(int m, int n) {
        if (iszero(m)) {
            return iszero(n);
        } else if (iszero(n)) {
            return false;
        } else {
            return eq(pi(m), pi(n));
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter two numbers: ");
        int m = scanner.nextInt();
        int n = scanner.nextInt();
        if (eq(m, n)) {
            System.out.println(m + " and " + n + " are equal.");
        } else {
            System.out.println(m + " and " + n + " are not equal.");
        }
    }
}

File: ToC_2026/8/question4/Rust/2.rs
use std::io;

fn iszero(m: i32) -> bool { m == 0 }
fn pi(m: i32) -> i32 { if m == 0 { 0 } else { m - 1 } }

fn eq(m: i32, n: i32) -> bool {
    if iszero(m) {
        iszero(n)
    } else if iszero(n) {
        false
    } else {
        eq(pi(m), pi(n))
    }
}

fn main() {
    println!("Enter two numbers: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let nums: Vec<i32> = input.split_whitespace().map(|s| s.parse().unwrap()).collect();
    
    if eq(nums[0], nums[1]) {
        println!("{} and {} are equal.", nums[0], nums[1]);
    } else {
        println!("{} and {} are not equal.", nums[0], nums[1]);
    }
}